crates/aquascope/src/analysis/permissions/flow.rs

```diff
@@ -0,0 +1,501 @@
+//! Region flow analysis for lifetime errors.
+//!
+//! Answer queries of the form, can region 1 flow into region 2?
+//!
+//! 1. A region R is abstract (φ) if there exists a flow from an abstract source to R.
```

Comment:
I think we should stick to the Oxide definitions here. A region is abstract if it's a parameter to the function, and it's concrete otherwise.